data/mods/ssb/abilities.ts

```diff
@@ -196,887 +472,1759 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
+				disabled: false,
+				used: false,
+				virtual: true,
+			};
+			move = this.dex.getMove(typeMap[newTypes[1]]);
+			pokemon.moveSlots[2] = {
+				move: move.name,
+				id: move.id,
+				pp: move.pp,
+				maxpp: move.pp,
+				target: move.target,
+				disabled: false,
+				used: false,
+				virtual: true,
+			};
+		},
+		isNonstandard: "Custom",
+		gen: 8,
+	},
+
+	// Finland
+	windingsong: {
+		desc: "If this Pokemon's species is Alcremie, it alternates one of its moves between two different options at the end of each turn, depending on the forme of Alcremie.",
+		shortDesc: "Alcremie: alternates between moves each turn.",
+		name: "Winding Song",
+		isPermanent: true,
+		onResidual(pokemon) {
+			if (pokemon.species.baseSpecies !== 'Alcremie') return;
+			let coolMoves = [];
+			if (pokemon.species.forme === 'Lemon-Cream') {
+				coolMoves = ['Reflect', 'Light Screen'];
+			} else if (pokemon.species.forme === 'Ruby-Swirl') {
+				coolMoves = ['Refresh', 'Destiny Bond'];
+			} else if (pokemon.species.forme === 'Mint-Cream') {
+				coolMoves = ['Light of Ruin', 'Sparkling Aria'];
 			} else {
-				pokemon.side.addSideCondition('reflect');
+				coolMoves = ['Infestation', 'Whirlwind'];
+			}
+			let oldMove;
+			let move;
+			if (pokemon.moves.includes(this.toID(coolMoves[0]))) {
+				oldMove = this.toID(coolMoves[0]);
+				move = this.dex.getMove(coolMoves[1]);
+			}	else if (pokemon.moves.includes(this.toID(coolMoves[1]))) {
+				oldMove = this.toID(coolMoves[1]);
+				move = this.dex.getMove(coolMoves[0]);
+			} else {
+				return;
 			}
+			if (!oldMove || !move) return;
+			const sketchIndex = pokemon.moves.indexOf(oldMove);
+			if (sketchIndex < 0) return false;
+			const sketchedMove = {
+				move: move.name,
+				id: move.id,
+				pp: (move.pp * 8 / 5),
+				maxpp: (move.pp * 8 / 5),
+				target: move.target,
+				disabled: false,
+				used: false,
+			};
+			pokemon.moveSlots[sketchIndex] = sketchedMove;
+			pokemon.baseMoveSlots[sketchIndex] = sketchedMove;
+			this.add('-message', `Finland changed its move ${this.dex.getMove(oldMove).name} to ${move.name}!`);
 		},
-		onModifyPriority(priority, pokemon, target, move) {
-			if (move && pokemon.activeTurns === 1) return priority + 1;
+		isNonstandard: "Custom",
+		gen: 8,
+	},
+
+	// frostyicelad
+	iceshield: {
+		desc: "This Pokemon can only be damaged by direct attacks. This Pokemon cannot lose its held item due to another Pokemon's attack.",
+		shortDesc: "Can only be damaged by direct attacks. Cannot lose its held item.",
+		name: "Ice Shield",
+		onDamage(damage, target, source, effect) {
+			if (effect.effectType !== 'Move') {
+				if (effect.effectType === 'Ability') this.add('-activate', source, 'ability: ' + effect.name);
+				return false;
+			}
+		},
+		onTakeItem(item, pokemon, source) {
+			if (this.suppressingAttackEvents(pokemon) || !pokemon.hp || pokemon.item === 'stickybarb') return;
+			if (!this.activeMove) throw new Error("Battle.activeMove is null");
```

Comment:
```suggestion
			if (!this.activeMove) throw new Error("Battle.activeMove is falsy");
```

Reply:
This is the error used through the sim for the same check, so it actually is specifically `null`

Reply:
`activeMove` has type `ActiveMove | null`, so it can't have any other falsy value.